memtable.cc

```diff
@@ -395,7 +396,8 @@ class scanning_reader final : public flat_mutation_reader::impl, private iterato
     std::optional<dht::partition_range> _delegate_range;
     flat_mutation_reader_opt _delegate;
     const io_priority_class& _pc;
-    const query::partition_slice& _slice;
+    lw_shared_ptr<query::partition_slice> _slice_ptr;
```

Comment:
This doesn't look like a killswitch.

Reply:
btw, killswitches are not encouraged. It means that if there's a problem, it can't be bisected to the patch that introduced the problem (just to the patch that lifts the switch). We can allow it, but it goes against the idea of granular commits.

Reply:
> This doesn't look like a killswitch.
I make another slice that doesn't have the `option::reversed`, even if the original one had. That way, Reversing is never true for partition_snapshot_reader.

Reply:
> btw, killswitches are not encouraged. It means that if there's a problem, it can't be bisected to the patch that introduced the problem (just to the patch that lifts the switch). We can allow it, but it goes against the idea of granular commits.

I did consider that problem, however I'm not sure whether it's encouraged to lump each change of (schema/order of traversal/switched bounds of ranges/conformance to the new assumption of higher-level readers that memtable reverses by itself), in just one commit. These changes can be hardly isolated, as they directly affect the control flow. Perhaps the commit introducing `rows_iter_type make_iterator(mutation_partition::rows_type::const_iterator it)` or `rows_iter_type make_iterator(mutation_partition::rows_type::const_iterator it)` could have been made separate, but then it feels that they lack required context.